core/commands/pin/remotepin.go

```diff
@@ -667,5 +666,32 @@ func getRemotePinServiceInfo(env cmds.Environment, name string) (url, key string
 	if !present {
 		return "", "", fmt.Errorf("service not known")
 	}
-	return service.Api.Endpoint, service.Api.Key, nil
+	endpoint, err = normalizeEndpoint(service.Api.Endpoint)
+	if err != nil {
+		return "", "", err
+	}
+	return endpoint, service.Api.Key, nil
+}
+
+func normalizeEndpoint(endpoint string) (string, error) {
+	uri, err := neturl.ParseRequestURI(endpoint)
```

Comment:
A couple questions:
1. Is it safe/expected for us to resolve `.` and `..` as gopath.Clean does, or is that really the job the HTTP server?
2. Should we use `ParseRequest` and error on fragments? Otherwise we end up escaping the `#` in the URI

Reply:
1. It is safe, could be done on either end. Doing it on our end makes things easier to debug, because normalized URL is stored in plain text config file.
2. Fragments are really not expected, won't happen "by mistake", so does not seem worth additional code: a remote service will simply return Error 404 for a missing file.